### PR TITLE
init piccolo

### PR DESCRIPTION
**Pull request by youngtaekiim**
A new version of PR#488
previous : https://github.com/eclipse-symphony/symphony/pull/488

### CODE REVIEW

File: api/pkg/apis/v1alpha1/providers/target/piccolo/piccolo.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+	defer observ_utils.CloseSpanWithError(span, &err)
+
+	sLog.InfofCtx(ctx, "  P (Piccolo Target): getting artifacts: %s - %s, traceId: %s", deployment.Instance.Spec.Scope, deployment.Instance.ObjectMeta.Name, span.SpanContext().TraceID().String())
+
+	ret := make([]model.ComponentSpec, 0)
+	for _, component := range references {
+		properties := component.Component.Properties
+		name := properties["workload.name"].(string)
+		workloadType := properties["workload.type"].(string)
+		req, err := http.NewRequest("GET", i.Config.Url+workloadType+"/"+name, nil)
```
Comment (Haishi2016) [inline api/pkg/apis/v1alpha1/providers/target/piccolo/piccolo.go:101]: The Get() method and Apply() method should be consistent. For example, the Get() method gets by the path "workload.type/workload.name", while the Apply() method posts to path component.Name, which isn't necessary the same with workload.name. If workload.name and component.name are the same, probably we should exclude workload.name in properties and use component.Name in Get()

**Review by Haishi2016 — CHANGES_REQUESTED** [review]

File: api/pkg/apis/v1alpha1/providers/target/piccolo/piccolo.go (hunk 1 above)
Comment (Haishi2016) [reply api/pkg/apis/v1alpha1/providers/target/piccolo/piccolo.go:101]: Also, the Apply() method doesn't use workload.type. Maybe workload.type should be included in the request properties?

File: api/pkg/apis/v1alpha1/providers/target/piccolo/piccolo.go [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+type PiccoloTargetProvider struct {
+	Config  PiccoloTargetProviderConfig
+	Context *contexts.ManagerContext
+}
+
+func PiccoloTargetProviderConfigFromMap(properties map[string]string) (PiccoloTargetProviderConfig, error) {
+	ret := PiccoloTargetProviderConfig{}
+	if v, ok := properties["name"]; ok {
+		ret.Name = v
+	}
```
Comment (Haishi2016) [inline api/pkg/apis/v1alpha1/providers/target/piccolo/piccolo.go:43]: Extract url from properties here. Throw COAError if url is not found in properties.

**Review by Haishi2016 — CHANGES_REQUESTED** [review]

**Comment by youngtaekiim** [comment]
I am sorry that I accidentally changed the repository and made a wrong commit fix (amend), which caused the existing PR to be closed.
I post response to existing your code reviews in https://github.com/eclipse-symphony/symphony/pull/503.

**Pull request closed without merging**